p2p/transport/quicreuse/tracer_metrics.go

```diff
@@ -85,7 +85,9 @@ func (c *aggregatingCollector) RemoveConn(id string) {
 
 var collector *aggregatingCollector
 
-func init() {
+var initMetricsOnce sync.Once
+
+func initMetrics() {
 	const (
 		direction = "direction"
 		encLevel  = "encryption_level"
```

Comment:
On L103 we're using the default prometheus Registerer. I think in general we should accept this as a parameter rather than hardcoded since we are a library.

No change needed now. If you agree, let's open an issue.